benchmark/output_formatter.cpp

```diff
@@ -0,0 +1,181 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "output_formatter.hpp"
+#include "random.hpp"
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+namespace ddwaf::benchmark {
+
+namespace {
+
+using output_fn_type = void (*)(std::ostream &, const settings &,
+    const std::map<std::string, std::vector<runner::test_result>> &);
+
+constexpr double MILLI = 1e3;
+constexpr double MICRO = 1e6;
+
+// NOLINTBEGIN(*-narrowing-conversions,*-magic-numbers)
+double percentile(const std::vector<uint64_t> &values, unsigned percentile)
+{
+    std::size_t size = values.size();
+    std::size_t index = ceil((size * percentile) / 100.0);
```

Comment:
usually you either do the closest rank or a linear interpolation. With `ceil` the value is biased.